packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1645,6 +1663,50 @@ class StackViewport extends Viewport implements IStackViewport {
           return;
         }
 
+        const pixelData = image.getPixelData();
+
+        // handle the case where the pixelData is a Float32Array
+        // CPU path cannot handle it, it should be converted to Uint16Array
+        // and via the Modality LUT we can display it properly
+        if (pixelData instanceof Float32Array) {
```

Comment:
This feels like a change that could be directly added to the Image class, rather than being in StackViewport, and potentially required for other Viewport types?

Reply:
This is for CPU, image class doesn't care about GPU or CPU